src/OpenLoco/src/Map/Track/TrackData.cpp

```diff
@@ -341,146 +344,603 @@ namespace OpenLoco::World::TrackData
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 34,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x114,
+            .flags = MiscFlags::sBendCurve | MiscFlags::oneSided,
+            .reverseTrackId = 40,
+            .reverseRotation = 2,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 34,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x114,
+            .flags = MiscFlags::sBendCurve | MiscFlags::oneSided,
+            .reverseTrackId = 39,
+            .reverseRotation = 2,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 34,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x114,
+            .flags = MiscFlags::sBendCurve | MiscFlags::oneSided,
+            .reverseTrackId = 38,
+            .reverseRotation = 2,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 34,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x25B,
+            .flags = MiscFlags::verySmallCurve | MiscFlags::oneSided,
+            .reverseTrackId = 43,
+            .reverseRotation = 0,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 26,
+            .sparkDirection = false },
+        MiscData{
+            .costFactor = 0x25B,
+            .flags = MiscFlags::verySmallCurve | MiscFlags::oneSided,
+            .reverseTrackId = 42,
+            .reverseRotation = 0,
+            .signalHeightOffsetLeft = 0,
+            .signalHeightOffsetRight = 0,
+            .compatibleFlags = TrackPieceFlags::oneSided,
+            .curveSpeedFraction = 0x0CCD,
+            .unkWeighting = 26,
+            .sparkDirection = false },
     };
 
-    uint16_t getRoadCompatibleFlags(size_t roadId)
+    const MiscData& getTrackMiscData(size_t trackId)
     {
-        return kRoadCompatibleFlags[roadId];
+        return _miscData[trackId];
     }
 
-    constexpr std::array<uint16_t, 10> kRoadCostFactor = {
-        0x100,
-        0x0C9,
-        0x0C9,
-        0x25B,
-        0x25B,
-        0x220,
-        0x220,
-        0x138,
-        0x138,
-        0x25B,
+    constexpr std::array<RoadMiscData, 10> _roadMiscData = {
```

Comment:
Do we have names for these to add as comments as well?

Reply:
We don't currently have names for road ids but I could probably come up with some names.